Author can attach an image to the article

ESteinman opened the pull request:
## PT-link: https://www.pivotaltracker.com/story/show/159759284

## User story:
```
As an Author,
In order to provide more value to my readers,
I would like to be able to attach an image to the article.
```

ESteinman committed:
new feature-file

ESteinman committed:
add Scenarios
and step-definition for attaching an image-file

ESteinman committed:
add image-association to article model

ESteinman committed:
add image-field to article form in view, add step-definition for I should see the [] image, and add scenario-step

ESteinman committed:
add @article.image.attach to articles controller

ESteinman committed:
rails active_storage:install

ESteinman committed:
add storage.yml-configs

ESteinman committed:
config active-storage :amazon

ESteinman committed:
add new gem aws-sdk-s3

ESteinman committed:
environment configuration

ESteinman committed:
config amazon to storage

ESteinman committed:
add gem aws-sdk-s3

ESteinman committed:
add image to articles-view

ESteinman committed:
add testing for attach a file and scenario for image

ESteinman committed:
unit test for attachment

ESteinman committed:
Merge branch 'Development' of https://github.com/CraftAcademy/D-pressed into attach_image

ESteinman committed:
encryption and active_storage into DB

ESteinman committed:
add association to article model

ESteinman committed:
add feature-testings and step-definition.

ESteinman committed:
add file field to article/new

ESteinman committed:
add image to article/show

ESteinman committed:
add old commits

ESteinman committed:
placement on each incoming tag on index.haml

ESteinman committed:
add encryption for dev and prod

ESteinman committed:
add and fix views

ESteinman committed:
fix step-definitions

ESteinman committed:
add Attachment - test green

ESteinman committed:
edit feature - author_attach_image_to_article.feature

ESteinman committed:
change configs

ESteinman committed:
edit article model

ESteinman committed:
edit index.haml

ESteinman committed:
add new gitignore-files

ESteinman committed:
edit feature

ESteinman committed:
add fixtures for unti testing

ESteinman committed:
controllers and views

ESteinman committed:
controllers and views

ESteinman committed:
config, specs and feature

tochman commented on .gitignore:
```diff
@@ -17,6 +18,7 @@
 /storage/*
 !/storage/.keep
 /coverage/
+/assets/images/
```
You want this in VCS

tochman commented on features/step_definitions/assertion_steps.rb:
```diff
@@ -13,4 +13,12 @@
\ No newline at end of file
+end
+When("I attach an image-file") do
+    attach_file('article_image', "#{::Rails.root}/spec/fixtures/dummy.jpeg")
```
Can’t see a file like the one you are pointing to...

ESteinman committed:
remove assets image from .gitignore

ESteinman committed:
delete app/assets/images/dummy.jpeg

ESteinman committed:
fix assertion_step

tochman committed:
fixes failing scenarios
changes way image is attached in controller by adding image to perimtted params
adds conditional image display in index page

tochman committed:
Merge branch 'attach_image' of github.com:ESteinman/D-pressed into attach_image

ESteinman committed:
solve merge conflicts

Andycutter reviewed (approved):
LGTM
